app/shared/utils/doLoginRedirect.ts

```diff
@@ -19,5 +25,6 @@ export const doLoginRedirect = async (router, user, campaign) => {
     ? '/volunteer-dashboard'
     : '/'
```

Comment:
Line 21 calls router.push(returnCookie), which returns void, but this is being assigned to redirectRoute and then used again on line 28. This is a logic error - the original code likely intended to return the URL string, not call router.push.


This should be

```
const redirectRoute: string = userHasRole(user, USER_ROLES.SALES)
  ? '/sales/add-campaign'
  : campaign
  ? await doPostAuthRedirect(campaign)
  : returnCookie || (
    status?.status === 'candidate'
    ? '/dashboard'
    : status?.status === 'volunteer'
    ? '/volunteer-dashboard'
    : '/'
  )

router.push(redirectRoute)

```

Reply:
Created a ticket for this to be prioritized outside of the TS -> JS conversion efforts: https://goodparty.clickup.com/t/90132012119/ENG-5907